### PR TITLE
[Feature] Add MLP Mixer Backbone.

### PR DESCRIPTION
**Pull request by xcnick**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add MLP-Mixer Image Classifier

## Modification

- Add MLP-Mixer backbone
- Add MLP-Mixer config files

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by mzr1996** [comment]
Thanks for your contribution, we will review this pull request recently.

File: mmcls/models/backbones/mlp_mixer.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        return getattr(self, self.norm1_name)
+
+    def forward(self, x):
+        x = self.patch_embed(x)
+
+        for layer in self.layers:
+            x = layer(x)
+
+        x = self.norm1(x).transpose(1, 2)
+        return x
```
Comment (0x4f5da2) [inline mmcls/models/backbones/mlp_mixer.py:218]: The output should be a tuple instead of a single tensor. Consider using `return (x, )` instead. 

Could you please further support `out_indices` to output the selected indices for more find grained output control?

**Review by 0x4f5da2 — CHANGES_REQUESTED** [review]

File: mmcls/models/backbones/mlp_mixer.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                nn.init.xavier_uniform_(m.weight)
+                nn.init.normal_(m.bias, std=1e-6)
+        for m in self.channel_mix.modules():
+            if isinstance(m, nn.Linear):
+                nn.init.xavier_uniform_(m.weight)
+                nn.init.normal_(m.bias, std=1e-6)
+
+    def forward(self, x):
+        out = self.norm1(x).transpose(1, 2)
+        x = x + self.token_mix(out).transpose(1, 2)
```
[the remark this answers is not in the document]
Comment (xcnick) [reply mmcls/models/backbones/mlp_mixer.py:89]: There is a `transpose` before identity connection. The forward interface of `FFN` does not support this.

File: mmcls/models/backbones/mlp_mixer.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        self.channel_mix = FFN(
+            embed_dims=embed_dims,
+            feedforward_channels=channels_mlp_dims,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
+
+    @property
+    def norm1(self):
```
[the remark this answers is not in the document]
Comment (mzr1996) [reply mmcls/models/backbones/mlp_mixer.py:69]: Since we use `norm_cfg` to build all kinds of norm layers, the `build_norm_layer` function will infer the abbreviation of these layers.
For example, `LayerNorm` -> `ln`, `BatchNorm` -> `bn`. It's helpful to set proper layer keys in checkpoints. So it's acceptable to use this naming format. 
Of course, using `norm1` & `norm2` directly is also acceptable. There is no strict naming convention.

File: mmcls/models/backbones/mlp_mixer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (0x4f5da2) [reply mmcls/models/backbones/mlp_mixer.py:89]: In this way, maybe you should set `add_identity=False` for `self.token_mix` to make the implementation correct.

File: mmcls/models/backbones/mlp_mixer.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        self.norm1_name, norm1 = build_norm_layer(
+            norm_cfg, embed_dims, postfix=1)
+        self.add_module(self.norm1_name, norm1)
+        self.token_mix = FFN(
+            embed_dims=num_tokens,
+            feedforward_channels=tokens_mlp_dims,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
```
Comment (0x4f5da2) [inline mmcls/models/backbones/mlp_mixer.py:55]:
```suggestion
            act_cfg=act_cfg,
            add_identity=False)
```

**Review by 0x4f5da2 — CHANGES_REQUESTED** [review]

File: mmcls/models/backbones/mlp_mixer.py (hunk 1 above)
Comment (xcnick) [reply mmcls/models/backbones/mlp_mixer.py]: Done. Added `out_indices` to support output control.

File: mmcls/models/backbones/mlp_mixer.py (hunk 4 above)
Comment (xcnick) [reply mmcls/models/backbones/mlp_mixer.py]: Done.

**Review by 0x4f5da2 — APPROVED** [review]
LGTM
